wallet/loader.go

```diff
@@ -240,6 +244,33 @@ func (l *Loader) LoadedWallet() (*Wallet, bool) {
 	return w, w != nil
 }
 
+// UnloadWallet stops the loaded wallet, if any, and closes the wallet database.
+// This returns ErrNotLoaded if the wallet has not been loaded with
+// CreateNewWallet or LoadExistingWallet.  The Loader may be reused if this
+// function returns without error.
+func (l *Loader) UnloadWallet() error {
+	defer l.mu.Unlock()
+	l.mu.Lock()
+
+	if l.wallet == nil {
+		return ErrNotLoaded
+	}
+
+	l.wallet.Stop()
+	l.wallet.WaitForShutdown()
+	// TODO: Closing the wallet database really should be done by Stop, but
```

Comment:
As discussed in IRC, it probably makes sense to keep the database close here in the loader since the caller might want to use a different backend and so the database needs to be passed into the wallet package like it is now in order to support that.

If the wallet package did the open it would be tied to whatever the wallet package wanted to use taking away the caller's flexibility.